frontend-react/src/utils/DiffCompare/AbstractDiffer.ts

```diff
@@ -0,0 +1,4 @@
+export type DifferMarkupResult = {
```

Comment:
smallest nit possible - if this is all that's going in this file you could maybe stash it in diff.ts

Reply:
I see your point. My logic for breaking it out was that the text differ and the json differ should have the same interface so they can be easily swapped. I should add a comment saying as much.

Reply:
Ah, I should make a whole interface and have them both implement it. I'll make this change.

Reply:
OK. I tried to change it to be an interface, but it cascaded into a bunch of weirdness since the function really does want json objects.

Look it over and see if it's good enough for now?